### PR TITLE
adding getBlockHeader method

### PR DESCRIPTION
**Pull request by jpikora-iov**

### CODE REVIEW

File: btc-transaction-helper.js [hunk 1]
```diff
@@ -295,6 +295,16 @@ class BtcTransactionHelper {
         return await this.nodeClient.getBlockCount();
     }
 
+    /**
+     * 
+     * @param {string} blockHash 
+     * @param {boolean} jsonEncoded defaults to true. If false, the block header will be returned as a hex string.
+     * @returns {BlockHeader | string} blockHeader
+     */
+    getBlockHeader(blockHash, jsonEncoded = false) {
```
Comment (jeremy-then) [inline btc-transaction-helper.js]: Why not set the `jsonEncoded` param as `true` by default? It is more useful that way.

File: index.d.ts [hunk 2]
```diff
@@ -41,8 +59,9 @@ export default interface BtcTransactionHelper {
     getTransaction(txHash: string): Promise<Transaction>;
     importAddress(address: string, label: string): Promise<null>;
     fundAddress(address: string, amountInBtc: number, mineBlock?: boolean = true): Promise<string>;
-    mine(blocks: number = 1): Promise<string>;
+    mine(blocks: number = 1): Promise<string[]>;
     getFee(): number;
     decodeBase58Address(address: string): string;
     getLatestBlockNumber(): Promise<number>;
+    getBlockHeader(blockHash: string, jsonEncoded: boolen = true): BlockHeader;
```
Comment (jeremy-then) [inline index.d.ts]: This says that it's true by default, but in the code it's false by default.

File: samples/transferBtc-samples.js [hunk 3]
```diff
@@ -314,4 +324,5 @@ const printP2PKHTransferHashWithNotEnoughForFees = async () => {
     await printP2PKHTransferHashWith0Change();
     await printP2PKHTransferHashWithChange();
     await printP2PKHTransferHashWithNotEnoughForFees();
+    await printBlockHeader()
```
Comment (jeremy-then) [inline samples/transferBtc-samples.js]: I don't think we need to add this sample here or to add this sample anywhere at all. Specially on this file which only shows samples of `transferBtc`. I think you can remove the changes from this file.

File: test/btc-transaction-helper.test.js [hunk 4]
```diff
@@ -486,4 +486,30 @@ describe('BtcTransactionHelper', () => {
...
+            nextblockhash: '0f7c230fc7326ddfa02e634a7165047d429db118ce60c74fb6428d079e0eedec'
+        }
+        
+        const btcTransactionHelper = new BtcTransactionHelper(config);
+        const getBlockHeaderStub = sinon.stub(btcTransactionHelper.nodeClient, 'getBlockHeader');
+        getBlockHeaderStub.resolves(blockHeader);
+        const result = await btcTransactionHelper.getBlockHeader(blockHeader.hash);
+        assert.deepEqual(result, blockHeader);
+        getBlockHeaderStub.restore();
+    })
```
Comment (jeremy-then) [inline test/btc-transaction-helper.test.js:514]: You are missing a test. Send an invalid `hash` to `getBlockHeader` and assert that it throws an exception `new Error('Block hash is required');`.

**Review by jeremy-then — COMMENTED** [review]
Left some comments.

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "btc-transaction-helper",
-  "version": "3.4.0",
+  "version": "3.5.0",
```
Comment (marcos-iov) [inline package.json:3]: Run `npm i` to update package-lock file

**Review by marcos-iov — APPROVED** [review]

**Review by nathanieliov — APPROVED** [review]
LGTM

**Pull request merged**